assignments/array-methods.js

```diff
@@ -55,29 +55,71 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
+// const fullName = [];
+
+// for(let i = 0; i < runners.length; i++){
+//     fullName.push(`${runners[i].first_name} ${runners[i].last_name}`);
+// }
+
+const fullName = runners.forEach(placeHolder => console.log(`${placeHolder.first_name} ${placeHolder.last_name}`));
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+// let allCaps = [];
+
+const allCaps = runners.map(placeHolder => placeHolder.first_name.toUpperCase());
+
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// let largeShirts = [];
+
+const largeShirts = runners.filter(placeHolder => placeHolder.shirt_size === "L" );
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+// let ticketPriceTotal = [];
+
+// const totalLandArea = cityData.reduce(function(banana, currentValue) {
+//   debugger;
+//   return banana + currentValue.population;
+  
+// }, 0);
+
+const ticketPriceTotal = runners.reduce((addLot, placeHolder) => addLot + placeHolder.donation, 0)
+
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Invoke the email and last name to return the full sentence "Thank you *last_name* for signing up with your email, *email*."
+
+const EndGreet = runners.forEach(client => console.log(`Thank you ${client.last_name} for signing up with your email, ${client.email}`));
+
+console.log(EndGreet);
 
 // Problem 2
+// Invoke the last client with the welcome, "Welcome *first_name* *Last_name* to our humble group!"
+
+const newPersonMsg = runners.filter(function(client) {
+
+    for(let i=0; i > runners.length; i++){
+     
+       if(client === runner.length-1){
+         
+         console.log(`Welcome ${client.first_name} ${client.last_name} to our humble group!`);
+       }
+    }
+});
+
+
+console.log(newPersonMsg); 
```

Comment:
I see where you are going with this and I think it can be done a little more simply. Is this what you were going for?
```suggestion
const person = runners.filter(person => person === runners[runners.length - 1]);

console.log(`Welcome ${person[0].first_name} ${person[0].last_name} to our humble group!`)
```
This would replace your existing filter.

This creates a new array with just the last runner object in the array. Then it consoles a message with the first and last name of the runner.